src/components/TankBar/TankBar.js

```diff
@@ -0,0 +1,326 @@
+    this.state.progressAnimated.addListener(({ value }) => this.updateLabelTransform(value));
+  }
+
+  updateLabelTransform = (value: number) => {
+    const { labelWidth, barWidth, sideButtonWidth } = this.state;
+    const barWidthPixels = barWidth * (value / 100);
+    const halfLabelWidth = labelWidth / 2;
+    let labelTransform = -(labelWidth / 2);
+    if (barWidthPixels <= halfLabelWidth) {
+      labelTransform = -barWidthPixels;
+    } else if (sideButtonWidth <= halfLabelWidth) {
+      labelTransform = -(halfLabelWidth - (sideButtonWidth - halfLabelWidth));
+    }
+    this.setState({ labelTransform });
+  };
+
+  firstAnimate = () => {
+    this.setState({ didFirstAnimation: true });
+    const { maxValue, currentValue } = this.props;
+    this.handleStatusValue(maxValue, currentValue);
+  };
+
+  handleStatusValue = (full: number, current: number) => {
+    let currentStatus = Math.floor((current * 100) / full);
+    if (current > full) currentStatus = 100;
+    this.animateProgress(currentStatus);
+  };
+
+  componentDidUpdate(prevProps: Props) {
+    const { maxValue, currentValue } = this.props;
+    if (prevProps.currentValue !== currentValue) {
+      this.handleStatusValue(maxValue, currentValue);
+    }
+  }
+
+  animateProgress = (newProgress: number) => {
+    Animated.spring(this.state.progressAnimated, {
+      toValue: newProgress,
+      duration: 400,
+      bounciness: 2,
+    }).start();
+  };
+
+  render() {
+    const {
+      progressAnimated,
+      barWidth,
+      sideButtonWidth,
+      labelTransform,
+      didFirstAnimation,
+    } = this.state;
+    const {
+      maxValue,
+      currentValue,
+      currentValueFormatted,
+      topupAction,
+      topUpLoading,
+    } = this.props;
+
+    return (
+      <Wrapper>
+        <Row style={{ marginBottom: spacing.large, marginTop: spacing.small }}>
+          <Title>PLR Tank</Title>
+        </Row>
+        <Row style={{ alignItems: 'flex-start' }}>
+          <ProgressBarSide>
+            <ProgressBarWrapper
+              onLayout={(e) => {
+                const { width } = e.nativeEvent.layout;
+                this.setState({ barWidth: width });
+              }}
+            >
+              <AnimatedProgress
+                style={{
+                  width: progressAnimated.interpolate({
+                    inputRange: [0, 100],
+                    outputRange: ['0%', '100%'],
+                  }),
+                }}
+                isFull={maxValue === currentValue || currentValue > maxValue}
+              />
+            </ProgressBarWrapper>
+          </ProgressBarSide>
+          <SideButton
+            onLayout={(e) => {
+              const { width } = e.nativeEvent.layout;
+              this.setState({ sideButtonWidth: width });
+            }}
+          >
+            <Value color={baseColors.darkGray}>{maxValue}</Value>
+            <Value style={{ marginLeft: 4 }}>{PPN_TOKEN}</Value>
+            { /* Uncomment when needed
```

Comment:
Can you update the comment on what it actually does (to know if we need it when we need it)?

Reply:
It's for a toggling between PLR and ETH, it's not yet required. I'll just remove it (and all other related parts you mention below) for now - not that difficult to implement when needed. Thanks!